src/main/java/org/nantipov/utils/wordhugopress/config/job/convertposts/ConvertPostsReaderConfig.java

```diff
@@ -25,25 +25,55 @@
 @Configuration
 public class ConvertPostsReaderConfig {
 
-    private static final String SQL_QUERY_POSTS =
-            "SELECT\n" +
+    private static final Set<String> DRAFT_STATUSES = ImmutableSet.of("draft", "auto-draft");
+
+    static final String BEAN_NAME = "postsReader";
+
+    private final SourcesSettings sourcesSettings;
+    
+    public ConvertPostsReaderConfig(SourcesSettings sourcesSettings) {
+        this.sourcesSettings = sourcesSettings;
+    }
+    
+    private static String TABLE_PREFIX;
+
+    private static String SQL_QUERY_POSTS; 
+
+    @Bean(BEAN_NAME)
+    public ItemReader<Post> reader() {
+        return CompositeItemReader.of(
+                sourcesSettings.getSources()
+                               .entrySet()
+                               .stream()
+                               .map(entry -> reader(entry.getKey(), entry.getValue()))
+                               .collect(Collectors.toList())
+        );
+    }
+
+    private ItemReader<Post> reader(String sourceName, SourcesSettings.Source source) {
+        if (source.getTags() != null && !source.getTags().isEmpty())
+            TABLE_PREFIX = source.getWordpressTablePrefix();
+        else
+            TABLE_PREFIX = "wp";
+        
+        SQL_QUERY_POSTS = "SELECT\n" +
```

Comment:
Method looks too big now. Maybe construction of the query could be moved to separate method?

Reply:
I agree to move this query in separate method